### PR TITLE
Task item master

### PR DESCRIPTION
**Pull request by nmimsnikhil**
ref:
- https://app.asana.com/0/1192407899754728/1196782244694132
- https://app.asana.com/0/1192407899754728/1196782244694136

item_master

### CODE REVIEW

File: erpnext/stock/doctype/item/item.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ class Item(WebsiteGenerator):
 	)
 
 	def onload(self):
+		# print(self.as_dict())
```
Comment (hrwX) [inline erpnext/stock/doctype/item/item.py]:
```suggestion
```

File: erpnext/stock/doctype/item/item.js [hunk 2]
```diff
@@ -5,6 +5,14 @@ frappe.provide("erpnext.item");
 
 frappe.ui.form.on("Item", {
 	setup: function(frm) {
+		frm.make_methods = {
+			'Purchase Order': () => {
+				frappe.model.open_mapped_doc({
+					method: 'erpnext.stock.doctype.item.item.make_purchase_order_item',
+					frm: frm
+				})
```
Comment (hrwX) [inline erpnext/stock/doctype/item/item.js]: - Do not need to do a make a frappe call
- Use frappe.new_doc here

File: erpnext/stock/doctype/item/item.py [hunk 3]
```diff
@@ -1132,3 +1133,10 @@ def toggle_variants_website_display(item_name, value):
 def on_doctype_update():
 	# since route is a Text column, it needs a length for indexing
 	frappe.db.add_index("Item", ["route(500)"])
+
+@frappe.whitelist()
+def make_purchase_order_item(source_name, target_doc=None):
+	item= frappe.get_doc("Item", {'item_code': source_name})
+	doc = frappe.new_doc("Purchase Order")
+	doc.supplier = item.item_defaults[0].default_supplier
+	return doc
```
Comment (hrwX) [inline erpnext/stock/doctype/item/item.py]:
```suggestion

```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/item/item.js (hunk 2 above)
Comment (nmimsnikhil) [reply erpnext/stock/doctype/item/item.js]: if i am using new_doc here it's not fetching 'item' in 'purchase order'.

**Review by hrwX — DISMISSED** [review]

**Review by hrwX — APPROVED** [review]

**Pull request merged**